omero_parade/csv_filters/omero_filters.py

```diff
@@ -0,0 +1,206 @@
+    Replace any non alpha-numeric character or . with underscore.
+
+    This allows us to use the filter name in URL to load filter script
+    """
+    w = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890.'
+    return "".join([l if l in w else '_' for l in name])
+
+
+def get_filters(request, conn):
+    obj_type = None
+    obj_id = None
+    for o in OBJ_TYPES:
+        if request.GET.get(o) is not None:
+            obj_id = request.GET.get(o)
+            obj_type = o
+            break
+
+    csv_anns = get_csv_annotations(conn, obj_type, obj_id)
+    # Use name_to_word so that filter can be used in URL
+    csv_names = [name_to_word(ann.file.name.val) for ann in csv_anns]
+    return csv_names
+
+
+def get_image_well_ids(conn, plate_id):
+    """Return dict of {image_id: well_id} for plate."""
+
+    img_to_well = {}
+    well_to_img = get_well_image_ids(conn, plate_id)
+    for well_id, img_id in well_to_img.items():
+        img_to_well[img_id] = well_id
+    return img_to_well
+
+
+def get_script(request, script_name, conn):
+    """Return a JS function to filter images by various params."""
+    obj = None
+    obj_type = None
+    for t in OBJ_TYPES:
+        if request.GET.get(t) is not None:
+            obj_id = int(request.GET.get(t))
+            obj = conn.getObject(t, obj_id)
+            obj_type = t
+            break
+
+    if obj is None:
+        return JsonResponse(
+            {'Error': 'Specify object e.g. ?project=1'})
+
+    # find csv file by name
+    file_ann = None
+    for ann in obj.listAnnotations():
+        if (ann.OMERO_TYPE == FileAnnotationI):
+            if name_to_word(ann.file.name.val) == script_name:
+                file_ann = ann
+
+    if file_ann is None:
+        return JsonResponse({'Error': 'csv file %s not found' % script_name})
+
+    # Prepare to find Well ID if needed
+    well_ids = {}
+    if obj_type == 'plate':
+        well_ids = get_image_well_ids(conn, obj_id)
+
+    # read file
+    chs = [ch.decode() for ch in file_ann.getFile().getFileInChunks()]
+    text = "".join(chs)
+    csv_reader = csv.reader(StringIO(text), delimiter=',')
+
+    # process each row, adding values to each column...
+    column_names = None
+    column_data = {}
+    for row in csv_reader:
+        # First row is column names:
+        if column_names is None:
+            column_names = [name.strip() for name in row]
+            # Batch_ROI_Export uses 'image_id'
+            if 'Image' not in column_names and 'image_id' in column_names:
+                column_names[column_names.index('image_id')] = 'Image'
+            for name in column_names:
+                column_data[name] = []
+            continue
+
+        # ignore empty rows
+        if not any(row):
+            continue
+
+        for name, value in zip(column_names, row):
+            column_data[name].append(value.strip())
+
+    # If Plate, we want Well IDs
+    if obj_type == 'plate' and 'Image' in column_names:
```

Comment:
This seems very strict for the name ``Image`` I can imagine someone using ``image`` and not having anything

Reply:
Fixed in a9c4b67